src/flamegraph/flamegraph.js

```diff
@@ -123,17 +164,18 @@ function g_to_func(e) {
 function update_text(e) {
     var r = find_child(e, "rect");
     var t = find_child(e, "text");
-    var w = parseFloat(r.attributes.width.value) -3;
-    var txt = find_child(e, "title").textContent.replace(/\\([^(]*\\)\$/,"");
-    t.attributes.x.value = parseFloat(r.attributes.x.value) + 3;
+    var framesWidth = frames.width.baseVal.value;
+    var w = parseFloat(r.attributes.width.value) * framesWidth / 100 - 3;
+    var txt = find_child(e, "title").textContent.replace(/\([^(]*\)$/,"");
+    t.attributes.x.value = format_percent((parseFloat(r.attributes.x.value) + (100 * 3 / framesWidth)));
     // Smaller than this size won't fit anything
     if (w < 2 * fontsize * fontwidth) {
         t.textContent = "";
         return;
     }
     t.textContent = txt;
     // Fit in full text width
-    if (/^ *\$/.test(txt) || t.getSubStringLength(0, txt.length) < w)
+    if (/^ *\$/.test(txt) || t.getComputedTextLength() < w)
```

Comment:
What's the reason for this change?

Reply:
I was getting errors because sometimes `txt.length` was `0`. We want the whole text here anyway.